#! /bin/sh /usr/share/dpatch/dpatch-run
## r1352.dpatch by Graham Dumpleton <graham.dumpleton@gmail.com>
##
## DP: Fixed bug where was decrementing Python object reference count on
## DP: NULL pointer, causing a crash. This was possibly only occuring in
## DP: embedded mode and only where closure of remote client connection was
## DP: detected before any request content was read. The issue may have been
## DP: more prevalent for a HTTPS connection from client

@DPATCH@
diff -urNad mod-wsgi-2.5~/mod_wsgi.c mod-wsgi-2.5/mod_wsgi.c
--- mod-wsgi-2.5~/mod_wsgi.c	2009-05-11 13:15:20.000000000 +0200
+++ mod-wsgi-2.5/mod_wsgi.c	2009-08-28 23:28:18.541018240 +0200
@@ -1540,7 +1540,6 @@
 
         if (n == -1) {
             PyErr_SetString(PyExc_IOError, "request data read error");
-            Py_DECREF(result);
             return NULL;
         }
 
